test/unit/sourceset.test.js

```diff
@@ -276,15 +300,263 @@ QUnit[qunitFn]('sourceset', function(hooks) {
         done();
       });
+
+      this.player.src([this.sourceOne, this.sourceTwo]);
     });
 
-    QUnit.test('no source', function(assert) {
+    QUnit.test('mediaEl.src = ...;', function(assert) {
       const done = assert.async();
 
-      this.player = videojs(this.mediaEl, {
-        enableSourceset: true
+      this.player = videojs(this.mediaEl, {enableSourceset: true});
+
+      this.player.one('sourceset', (e) => {
+        validateSource(assert, this.player, [this.testSrc]);
+        done();
+      });
+
+      this.player.tech_.el_.src = this.testSrc.src;
+    });
+
+    QUnit.test('mediaEl.setAttribute("src", ...)"', function(assert) {
+      const done = assert.async();
+
+      this.player = videojs(this.mediaEl, {enableSourceset: true});
+
+      this.player.one('sourceset', (e) => {
+        validateSource(assert, this.player, [this.testSrc]);
+        done();
+      });
+
+      this.player.tech_.el_.setAttribute('src', this.testSrc.src);
+    });
+
+    const appendTypes = [
+      {name: 'appendChild', fn: (el, obj) => el.appendChild(obj)},
+      {name: 'innerHTML', fn: (el, obj) => {el.innerHTML = obj.outerHTML;}}, // eslint-disable-line
+    ];
+
+    // ie does not support this and safari < 10 does not either
+    if (window.Element.prototype.append) {
+      appendTypes.push({name: 'append', fn: (el, obj) => el.append(obj)});
+    }
+
+    if (window.Element.prototype.insertAdjacentHTML) {
+      appendTypes.push({name: 'insertAdjacentHTML', fn: (el, obj) => el.insertAdjacentHTML('afterbegin', obj.outerHTML)});
+    }
+
+    appendTypes.forEach((appendObj) => {
+
+      QUnit.test(`<source> one source through ${appendObj.name}`, function(assert) {
+        const done = assert.async();
+
+        this.source = document.createElement('source');
+        this.source.src = this.testSrc.src;
+        this.source.type = this.testSrc.type;
+
+        this.player = videojs(this.mediaEl, {enableSourceset: true});
+
+        this.player.one('sourceset', (e) => {
+          validateSource(assert, this.player, [this.testSrc], false);
+          done();
+        });
+
+        // since the media el has no source, just appending will
+        // change the source without calling load
+        appendObj.fn(this.player.tech_.el_, this.source);
+      });
+
+      QUnit.test(`<source> one source through ${appendObj.name} and load`, function(assert) {
+        const done = assert.async();
+
+        this.totalSourcesets = 2;
+        this.source = document.createElement('source');
+        this.source.src = this.testSrc.src;
+        this.source.type = this.testSrc.type;
+
+        this.player = videojs(this.mediaEl, {enableSourceset: true});
+
+        this.player.one('sourceset', (e) => {
+          validateSource(assert, this.player, [this.testSrc], false);
+
+          this.player.one('sourceset', (e2) => {
+            // should be the same event
+            validateSource(assert, this.player, [this.testSrc], false);
```

Comment:
is this a test that we want to assert based on the value of `e.src` and `e2.src`? because otherwise this doesn't doesn't do anything beyond checking that two sourceset events fired.

Reply:
yeah your right I will revert this. This was changed in the `update player cache` branch, and validateSource actually takes the event in that one. I will put this back